File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/common/CheckboxTests.java
package com.epam.jdi.uitests.testing.unittests.tests.common;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.tests.common.dataProviders.CheckBoxDP;
import com.epam.jdi.uitests.win.winium.elements.common.CheckBox;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.io.IOException;
import java.lang.reflect.Method;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.METALS_AND_COLORS_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.metalsColorsPage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.checkAction;
import static org.testng.Assert.assertFalse;
import static org.testng.Assert.assertTrue;

/**
 * Created by Dmitry_Lebedev1 on 15/12/2015.
 */
public class CheckboxTests extends InitTests {

    public CheckBox checkBox() {
        return metalsColorsPage.cbWater;
    }

    @BeforeMethod
    public void before(final Method method) throws IOException {
        isInState(METALS_AND_COLORS_PAGE, method);
    }

    @Test
    public void checkSingleTest() throws InterruptedException {
        checkBox().check();
        checkAction("Water: condition changed to true");
    }

    @Test
    public void uncheckSingleTest() throws InterruptedException {
        checkBox().click();
        checkBox().uncheck();
        checkAction("Water: condition changed to false");
    }

    @Test
    public void isCheckTest() throws Exception {
        assertFalse(checkBox().isChecked());
        checkBox().click();
        assertTrue(checkBox().isChecked());
    }

    @Test
    public void multiUncheckTest() throws Exception {
        checkBox().click();
        checkBox().uncheck();
        checkBox().uncheck();
        checkAction("Water: condition changed to false");
    }

    @Test
    public void multiCheckTest() throws Exception {
        checkBox().check();
        checkBox().check();
        checkAction("Water: condition changed to true");
    }

    @Test
    public void clickTest() throws Exception {
        checkBox().click();
        checkAction("Water: condition changed to true");
        checkBox().click();
        checkAction("Water: condition changed to false");
    }

    @Test(dataProvider = "inputData", dataProviderClass = CheckBoxDP.class)
    public void setValue(String input, Boolean expected){
        if (!expected)
            checkBox().click();
        checkBox().setValue(input);
        String resMsg = "Water: condition changed to " + expected.toString();
        checkAction(resMsg);
    }

    @Test(dataProvider = "inputDataNothing2Changed", dataProviderClass = CheckBoxDP.class)
    public void setValueNothing2Changed(String input){
        checkBox().click();
        checkBox().setValue(input);
        checkAction("Water: condition changed to true");
        checkBox().click();
        checkBox().setValue(input);
        checkAction("Water: condition changed to false");
    }
}

File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/Tests/Common/CheckBoxTests.cs
using NUnit.Framework;
using static Epam.JDI.Core.Settings.JDISettings;
using static JDI_UIWebTests.UIObjects.TestSite;
using static JDI_UIWebTests.Tests.Complex.CommonActionsData;
using Epam.JDI.Core.Interfaces.Common;
using JDI_UIWebTests.DataProviders;
using Assert = JDI_Matchers.NUnit.Assert;

namespace JDI_UIWebTests.Tests.Common
{
    public class CheckBoxTests
    {

        private ICheckBox _checkBoxWater = MetalsColorsPage.CbWater;

        [SetUp]
        public void SetUp()
        {
            Logger.Info("Navigating to Metals and Colors page.");
            MetalsColorsPage.Open();
            MetalsColorsPage.CheckTitle();
            MetalsColorsPage.IsOpened();
            Logger.Info("Setup method finished");
            Logger.Info("Start test: " + TestContext.CurrentContext.Test.Name);
        }

        [Test]
        public void CheckSingleTest() 
        {
            _checkBoxWater.Check();            
            CheckAction("Water: condition changed to true");
        }

        [Test]
        public void UncheckSingleTest()
        {
            _checkBoxWater.Click();
            _checkBoxWater.Uncheck();
            CheckAction("Water: condition changed to false");
        }

        [Test]
        public void IsCheckTest()
        {
            Assert.IsFalse(_checkBoxWater.IsChecked());
            _checkBoxWater.Click();
            Assert.IsTrue(_checkBoxWater.IsChecked());
        }

        [Test]
        public void MultipleUncheckTest()
        {
            _checkBoxWater.Click();
            _checkBoxWater.Uncheck();
            _checkBoxWater.Uncheck();
            CheckAction("Water: condition changed to false");
        }

        [Test]
        public void ClickTest()
        {
            _checkBoxWater.Click();
            CheckAction("Water: condition changed to true");
            _checkBoxWater.Click();
            CheckAction("Water: condition changed to false");
        }

        [Test]
        [TestCaseSource(typeof(CheckBoxProvider), nameof(CheckBoxProvider.InputData))]
        public void SetValueTest(string value, bool expected)
        {
            if (!expected)
            {
                _checkBoxWater.Click();
            }
            _checkBoxWater.Value = value;
            string resultMsg = "Water: condition changed to " + expected.ToString().ToLower();
            CheckAction(resultMsg);
        }

        [Test]
        [TestCaseSource(typeof(CheckBoxProvider), nameof(CheckBoxProvider.InputValue))]
        public void SetWrongValueTest(string value)
        {
            _checkBoxWater.Click();
            _checkBoxWater.Value = value;
            CheckAction("Water: condition changed to true");
            _checkBoxWater.Click();
            _checkBoxWater.Value = value;
            CheckAction("Water: condition changed to false");
        }
    }
}
